engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterLaunchEngine.m

```diff
@@ -0,0 +1,52 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterLaunchEngine.h"
+
+@interface FlutterLaunchEngine () {
+  BOOL _didTakeEngine;
+  FlutterEngine* _engine;
+}
+@end
+
+@implementation FlutterLaunchEngine
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self->_didTakeEngine = NO;
+  }
+  return self;
+}
+
+- (FlutterEngine*)engine {
+  if (!_didTakeEngine && !_engine) {
+    // `allowHeadlessExecution` is set to `YES` since that has always been the
```

Comment:
Should we issue a warning to users to migrate to FlutterPluginRegistrant?

Reply:
We can add to here, and to the other PR (tools change) that scans the code usage.

Reply:
Would the default AppDelegate template trigger this? I think it does? If so, probably don't want to put a warning because that will hit basically everyone.

```objc
@main
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
      
    GeneratedPluginRegistrant.register(with: self)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
}
```

Reply:
I added an issue to introduce it.  It should point to the migration guide which hasn't landed yet: https://github.com/flutter/flutter/issues/169678

Reply:
> Would the default AppDelegate template trigger this? I think it does? If so, probably don't want to put a warning because that will hit basically everyone.

Yea, good call.  I'll note that in the issue.  I don't know where I've mentioned it but I think after this has all landed we should move GeneratedPluginRegistrant calls to use the new API and make GeneratedPluginRegistrant inherit from FlutterPluginRegistrant.  That's down the line though.